src/main/java/reposense/RepoSense.java

```diff
@@ -62,7 +58,7 @@ public static void main(String[] args) {
                 return;
             } else if (cliArguments instanceof ConfigCliArguments) {
                 configs = getRepoConfigurations((ConfigCliArguments) cliArguments);
-                reportConfig = getReportConfigurations((ConfigCliArguments) cliArguments);
```

Comment:
If configuration has to be set and parsed twice, does it mean that this code here is run twice? Why so?

Reply:
This PR needs to be viewed in the context of #1501. The configuration parsing would have to be done twice if we kept this function inside RepoSense.java.

Putting them together here allows us to get information from both sources, then do a final output (something like a parse-parse-set instead of parse-set-parse-set).

For this function in particular, the configuration does not get parsed twice. It’s just retrieving the ReportConfiguration object, which has already been prepared earlier in ArgsParser.parse().